Sources/Commands/SwiftTestTool.swift

```diff
@@ -1139,3 +1140,141 @@ private extension SwiftTool {
         return parameters
     }
 }
+
+extension SwiftTestTool {
```

Comment:
should we define a new target "Test" or "PackageTest" (the build one is call "Build"), move this code there, and then reuse from both call sites?

Reply:
or even to workspace if we must

Reply:
the other alternative (maybe better) it to move all this code into SwiftPackageTool which handles plugins already (so that the fact doTest returns PluginInvocationTestResult is less leaky), and move the shared utilities (eg getTestSuites) to some static helper that all tools can use, e.g. TestUtilities that lives in the tools area or even defined in SwiftTool.swift

Reply:
That seems reasonable.  That would be short of the rewrite that this SwiftTestTool needs but still not add too much to the problem.

Reply:
Factored this out into a separate source file and moved the main logic back into SwiftPackageTool.  Future improvements would be to move it down into its own module.